src/gui/widgets/Fader.cpp

```diff
@@ -315,154 +284,207 @@ void Fader::updateTextFloat()
 }
 
 
-inline int Fader::calculateDisplayPeak( float fPeak )
-{
-	int peak = static_cast<int>(m_back.height() - (fPeak / (m_fMaxPeak - m_fMinPeak)) * m_back.height());
-
-	return qMin(peak, m_back.height());
-}
-
-
 void Fader::paintEvent( QPaintEvent * ev)
 {
 	QPainter painter(this);
 
-	// Draw the background
-	painter.drawPixmap(ev->rect(), m_back, ev->rect());
-
 	// Draw the levels with peaks
-	if (getLevelsDisplayedInDBFS())
-	{
-		paintDBFSLevels(ev, painter);
-	}
-	else
-	{
-		paintLinearLevels(ev, painter);
-	}
+	// TODO LMMS Compressor, EQ and Delay still use linear displays...
+	paintLevels(ev, painter, !getLevelsDisplayedInDBFS());
 
 	// Draw the knob
-	painter.drawPixmap(0, knobPosY() - m_knob.height(), m_knob);
+	painter.drawPixmap((width() - m_knob.width()) / 2, knobPosY() - m_knob.height(), m_knob);
 }
 
-void Fader::paintDBFSLevels(QPaintEvent * ev, QPainter & painter)
+void Fader::paintLevels(QPaintEvent * ev, QPainter & painter, bool linear)
 {
-	int height = m_back.height();
-	int width = m_back.width() / 2;
-	int center = m_back.width() - width;
+	std::function<float(float value)> mapper = [this](float value) { return ampToDbfs(qMax<float>(0.0001, value)); };
+
+	if (linear)
+	{
+		mapper = [this](float value) { return value; };
+	}
 
-	float const maxDB(ampToDbfs(m_fMaxPeak));
-	float const minDB(ampToDbfs(m_fMinPeak));
+	float const mappedMinPeak(mapper(m_fMinPeak));
+	float const mappedMaxPeak(mapper(m_fMaxPeak));
+	float const mappedPeakL(mapper(m_fPeakValue_L));
+	float const mappedPeakR(mapper(m_fPeakValue_R));
+	float const mappedPersistentPeakL(mapper(m_persistentPeak_L));
+	float const mappedPersistentPeakR(mapper(m_persistentPeak_R));
+	float const mappedUnity(mapper(1.f));
 
-	// We will need to divide by the span between min and max several times. It's more
-	// efficient to calculate the reciprocal once and then to multiply.
-	float const fullSpanReciprocal = 1 / (maxDB - minDB);
+	painter.save();
 
+	QRect const baseRect = rect();
 
-	// Draw left levels
-	float const leftSpan = ampToDbfs(qMax<float>(0.0001, m_fPeakValue_L)) - minDB;
-	int peak_L = height * leftSpan * fullSpanReciprocal;
-	QRect drawRectL( 0, height - peak_L, width, peak_L ); // Source and target are identical
-	painter.drawPixmap(drawRectL, m_leds, drawRectL);
-
-	float const persistentLeftPeakDBFS = ampToDbfs(qMax<float>(0.0001, m_persistentPeak_L));
-	int persistentPeak_L = height * (1 - (persistentLeftPeakDBFS - minDB) * fullSpanReciprocal);
-	// the LED's have a  4px padding and we don't want the peaks
-	// to draw on the fader background
-	if( persistentPeak_L <= 4 )
-	{
-		persistentPeak_L = 4;
-	}
-	if( persistentLeftPeakDBFS > minDB )
+	int const height = baseRect.height();
+
+	int const margin = 1;
+	int const distanceBetweenMeters = 2;
+
+	int const numberOfMeters = 2;
+	int const meterWidth = (baseRect.width() - 2 * margin - distanceBetweenMeters * (numberOfMeters - 1)) / numberOfMeters;
```

Comment:
I feel we should be explicit with parentheses in math like this that we do in fact mean `(2 * margin)` and `(distanceBetweenMeters * ...)`. Just a thought.

Reply:
Fixed with commit 2bec339fb8a. It introduces intermediate variables to make the intentions a bit clearer. This also reduces the number of parentheses.